### PR TITLE
macOS Support

### PR DESCRIPTION
**Pull request by alexrabin-sentracam**
Addresses https://github.com/csdcorp/speech_to_text/issues/231

This pr adds macOS support with a HUGE caveat.

This functionality only works when running the app directly from Xcode. There is a known issue with flutter that the macOS app crashes when requesting permissions in VSCode https://github.com/flutter/flutter/issues/70374.

I leave this up to the author on if they want to merge this or not. But the functionality does exist.

### CODE REVIEW

**Review by sowens-csd — COMMENTED** [review]
I'm excited about MacOS support, thanks for doing the work. I'd definitely like to merge. I've only had a quick look at this PR as it is quite large but I don't understand yet how it provides MacOS support. Could you point me at the two or three relevant files so I can review? So far I mostly see file deletions but not corresponding additions.

**Comment by alexrabin-sentracam** [comment]
@sowens-csd I was able to implement macOS support by using the shared Darwin approach as mentioned in the flutter docs: https://docs.flutter.dev/packages-and-plugins/developing-packages#shared-ios-and-macos-implementations. Both iOS and macOS use the same code with some slight differences based off their platforms.

Take a look at the speech_to_text/darwin/Classes/SpeechToTextPlugin.swift file for the implementation.

**Review by sowens-csd — COMMENTED** [review]
Thanks, very helpful. I got caught by GitHub not showing large diffs by default. Going through it now.

**Comment by alibad** [comment]
@sowens-csd any plans to merge and launch it in a new version of the flutter package?

Are you aware of speech_to_text_macos?

**Comment by alexrabin-sentracam** [comment]
@alibad "speech_to_text_macos" is no longer needed because of the shared darwin implementation

File: examples/audio_player_interaction/macos/Runner/DebugProfile.entitlements [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>com.apple.security.app-sandbox</key>
+	<true/>
+	<key>com.apple.security.cs.allow-jit</key>
+	<true/>
+	<key>com.apple.security.device.audio-input</key>
+	<true/>
+	<key>com.apple.security.network.server</key>
```
Comment (alibad) [inline examples/audio_player_interaction/macos/Runner/DebugProfile.entitlements:11]: should this also include

	<key>com.apple.security.network.client</key>
	<true/>

File: examples/audio_player_interaction/macos/Podfile [hunk 2]
```diff
@@ -0,0 +1,43 @@
+platform :osx, '10.14'
```
Comment (alibad) [inline examples/audio_player_interaction/macos/Podfile:1]: so it is compatible with MacOS 10.14?

File: examples/audio_player_interaction/ios/Podfile [hunk 3]
```diff
@@ -1,5 +1,5 @@
 # Uncomment this line to define a global platform for your project
-# platform :ios, '11.0'
+# platform :ios, '12.0'
```
Comment (alibad) [inline examples/audio_player_interaction/ios/Podfile:2]: was this iOS upgrade intended?

File: examples/audio_player_interaction/macos/Podfile (hunk 2 above)
Comment (alexrabin-sentracam) [reply examples/audio_player_interaction/macos/Podfile:1]: The plugin can be built on macOS 10.13 and above but the speech recognition code only works on macOS 10.15 and above.

When I generated the macOS project the pod file automatically set the platform as 10.14

**Review by alibad — COMMENTED** [review]
What kind of testing was done to validate the changes in this PR work, on iOS and MacOS?

File: examples/audio_player_interaction/macos/Runner/DebugProfile.entitlements (hunk 1 above)
Comment (alexrabin-sentracam) [reply examples/audio_player_interaction/macos/Runner/DebugProfile.entitlements:11]: To run the debug version of the audio_player_interaction on macOS the network.client. key is not needed.

File: examples/audio_player_interaction/ios/Podfile (hunk 3 above)
Comment (alexrabin-sentracam) [reply examples/audio_player_interaction/ios/Podfile:2]: Probably not. When I generated the macOS project in the example it updated the platform numbers. DO you want me to revert it?

**Comment by alexrabin-sentracam** [comment]
@alibad I ran the example projects on iOS and macOS on my Mac. I made sure text was returned from the plugin.

File: examples/audio_player_interaction/ios/Podfile (hunk 3 above)
Comment (alibad) [reply examples/audio_player_interaction/ios/Podfile:2]: I personally think it's fine, unless @sowens-csd has reasons not to

File: examples/audio_player_interaction/macos/Podfile (hunk 2 above)
Comment (alibad) [reply examples/audio_player_interaction/macos/Podfile:1]: so it's best be at 10.15?

File: examples/audio_player_interaction/macos/Runner/DebugProfile.entitlements (hunk 1 above)
Comment (alibad) [reply examples/audio_player_interaction/macos/Runner/DebugProfile.entitlements:11]: Even when speech recognition is not running locally on the device?

Comment (alexrabin-sentracam) [reply examples/audio_player_interaction/macos/Runner/DebugProfile.entitlements:11]: Correct, when I tested locally that key is not needed when running on device.

**Comment by alibad** [comment]
@sowens-csd can you please approve this change? I'd like to use it as soon as the new version comes out.

**Comment by sowens-csd** [comment]
@alexrabin-sentracam I've merged the PR and am trying to run the example. I've tried it from Xcode after a clean build and the app appears but clicking initialize does not appear to do anything. No errors in the log.  

So far I've been unable to get the debugger to stop on any lines inside the .swift. Any tips on how you tested that might help?

**Comment by alexrabin-sentracam** [comment]
@sowens-csd I pulled the latest code and I was able to get it to work correctly in Xcode. What macOS version are you on? I am on macOS 14.5

**Comment by alexrabin-sentracam** [comment]
@sowens-csd make sure to run

`flutter pub get` and `cd macos && pod install` It's possible your local is still pointing to the old speech_to_text_macos plugin

**Comment by sowens-csd** [comment]
Thanks for the assist, working well now. I'll add that to the README in case anyone is as much of an idiot as I. Thanks so much for doing this work. It's going to be in the 7.0.0 release of the plugin.

**Comment by alexrabin-sentracam** [comment]
Happy to help!

**Comment by alibad** [comment]
Exciting! @sowens-csd when will the 7.0.0 release of the plugin take place? I'd like to pull it right away to test the MacOS behavior.

**Comment by alibad** [comment]
I just tested the beta. Before, I would get an initialize error. Now, the app crashes without any errors logged

**Comment by sowens-csd** [comment]
Did you run it from Xcode or VSCode? There's apparently a permissions problem that prevent Mac apps from working unless run from Xcode.

**Comment by alibad** [comment]
Yes, it works from there

**Comment by alexrabin-sentracam** [comment]
@sowens-csd you might want to mention higher up in the ReadMe that MacOS only works when being run from Xcode.

**Pull request merged**